fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryWrapper.java

```diff
@@ -107,6 +108,15 @@ private FDBDirectoryMergeScheduler(@Nonnull IndexMaintainerState state, int merg
         })
         @Override
         public synchronized void merge(final MergeSource mergeSource, final MergeTrigger trigger) throws IOException {
+            if (trigger == MergeTrigger.CLOSING) {
+                final IndexMergePolicy indexMergePolicy = state.store.getIndexMergePolicy();
+                if (!indexMergePolicy.isMergeDuringClose()) {
+                    // Here it is done for the record - if this store is being closed it is too late to use it as a messenger
+                    // (See the other code points that set this value)
+                    indexMergePolicy.setIndexNeedsMerge(true);
```

Comment:
Why are we doing this?

Reply:
Just in case the was an explicit "close" call. 
(I don't know if it's possible at all)